app/models/mixins/authentication_mixin.rb

```diff
@@ -12,6 +12,33 @@ def self.authentication_check_schedule
       assocs = zone.respond_to?(assoc) ? zone.send(assoc) : []
       assocs.each { |a| a.authentication_check_types_queue(:attempt => 1) }
     end
+
+    def self.queue_authentication_check(args, user_id, zone)
```

Comment:
@AparnaKarve I played around with it and it was around 7 seconds on average, with the shortest time being 4 seconds and the longest being 10 seconds